### PR TITLE
add cloudflare_managed_headers suppport

### PR DESCRIPTION
**Pull request by jafowler**
This PR adds support for the `zone/:id/managed_headers` endpoint. One weird thing about the `ListZoneManagedHeaders` is that it seems to be one of the only `List` functions that doesn't return a list. So i opted to convert the response into a list to make the transformations work.

### CODE REVIEW

**Comment by jafowler** [comment]
@jacobbednarz want me to update the test data for the two failing tests?

File: testdata/terraform/cloudflare_managed_headers/test.tf
Comment (jacobbednarz) [inline testdata/terraform/cloudflare_managed_headers/test.tf:7]: i wouldn't expect the `enabled = false` ones to be outputted in TF.

**Comment by jacobbednarz** [comment]
yes please; they are fixes from upstream that have been pulled in.

File: testdata/terraform/cloudflare_managed_headers/test.tf
Comment (jafowler) [reply testdata/terraform/cloudflare_managed_headers/test.tf:7]: ah right, i update the code, just not the test, let me fix that.

**Comment by jafowler** [comment]
looks like @tamas-jozsa has the certificate pack changes in his branch. I'll leave those to avoid merge conflicts for him.

**Comment by tamas-jozsa** [comment]
> looks like @tamas-jozsa has the certificate pack changes in his branch. I'll leave those to avoid merge conflicts for him.

Feel free to add them here to unblock your PR. My PR still needs some work and I will do a rebase anyway.

**Comment by jafowler** [comment]
> > looks like @tamas-jozsa has the certificate pack changes in his branch. I'll leave those to avoid merge conflicts for him.
> 
> Feel free to add them here to unblock your PR. My PR still needs some work and I will do a rebase anyway.

Talked with @tamas-jozsa in order to get the tests and validation passing, i'd need to do the exact same work he is doing now. I will rebase after his stuff is merge.d

**Comment by jacobbednarz** [comment]
you should be able to rebase in the changes and be all good now.

**Comment by jafowler** [comment]
> you should be able to rebase in the changes and be all good now.

Done!

**Review by jacobbednarz — APPROVED** [review]

**Pull request merged**
